tests/qemu/test_qemu_backend.cpp

```diff
@@ -530,6 +530,37 @@ TEST_F(QemuBackend, ssh_hostname_returns_expected_value)
     EXPECT_EQ(machine.VirtualMachine::ssh_hostname(), expected_ip);
 }
 
+TEST_F(QemuBackend, gets_management_ip)
+{
+    mpt::StubVMStatusMonitor stub_monitor;
+    const std::string expected_ip{"10.10.0.35"};
+    NiceMock<mpt::MockDNSMasqServer> mock_dnsmasq_server{data_dir.path(), bridge_name, subnet};
+
+    ON_CALL(mock_dnsmasq_server, get_ip_for(_)).WillByDefault([&expected_ip](auto...) {
+        return mp::optional<mp::IPAddress>{expected_ip};
+    });
```

Comment:
Why the lambda? Should be able to just `....WillByDefault(Return(expected_ip))`.

And might as well EXPECT_CALL, no?

Reply:
Right, changed the lambda and added the `EXPECT_CALL`.

Reply:
Why did you keep the `ON_CALL` though? You have both `ON_CALL` and `EXPECT_CALL` now.

Reply:
Just read the difference between them. I was just confused on the meaning of `ON_CALL`. Changed.